ideas/2024.md

```diff
@@ -23,30 +23,57 @@ medium
 Prior efforts:
 there was some prior efforts in [nixpkgs#258250](https://github.com/NixOS/nixpkgs/pull/258250) but it is stuck for a few months now and also only adds function for legacy ip, it's also quite bare bones and we might want some more advanced things like a function to check if a ip matches a given subnet.
 
-### First class support for Crystal applications.
+### First class support for Crystal applications
 
 Currently, our support for applications written in crystal consists of 2 main pieces:
 
-1. `crystal2nix` for converting lock files created by the `shards` tool to nix
-2. the nixpkgs build infrastructure that supports a range of different means to build crystal applications (incl consuming the outputs of item 1)
+1. `crystal2nix` for converting lock files created by the `shards` tool to nix.
+2. the nixpkgs build infrastructure that supports a range of different means to build crystal applications (incl consuming the outputs of item 1).
    
-The existing support is very brittle and unpleasant to read and only support dependencies fetched via git. 
+The existing support is brittle and unpleasant to read and only supports dependencies fetched via git. 
 
-This project would be interesting as it touches seveal different types of projects:
+This project would be interesting as it touches several different types of projects:
 
 1. redesigning the build infrastructure to be easier to hack on (nixpkgs)
-2. implmenting the new build infrastructure (nixpkgs)
+2. implementing the new build infrastructure (nixpkgs)
 3. extending the `crystal2nix` tool to understand other VCSs (crystal2nix)
 
-Skills required/preferred:
-Crystal because crystal2nix is written in it.
+Skills preferred:
+a. nix and bash for the build infrastructure
+b. crystal or ruby for `crystal2nix` (if you know ruby, you know crystal)
 
 Possible Mentors:
 - [@peterhoeg](https://github.com/peterhoeg)
 
 Difficulty:
 easy
 
+Prior efforts:
+We have both working crystal build infrastructure as well as a working [crystal2nix](https://github.com/nix-community/crystal2nix), so you will not be starting from scratch.
+
+### Centralized on-failure reporting for failed systemd units
```

Comment:
everything below line 54 should probably be it's own commit for the systemd unit reporting idea.

Reply:
I can split it up, but the commit history here doesn't really matter, does it?